src/pal/inc/pal_endian.h

```diff
@@ -98,8 +98,9 @@ inline void SwapGuid(GUID *pGuid)
 #define VALPTR(x) VAL32(x)
 #endif
 
-#if defined(ALIGN_ACCESS) && !defined(_MSC_VER)
-
+#if defined(_TARGET_ARM_) && !defined(_MSC_VER)
+#define LOG2_PTRSIZE	2
```

Comment:
I would keep the existing code as is, and just defined the `ALIGN_ACCESS` macro. It will make it easy to define it for new platforms as necessary. Also, this should be just `_ARM_` not `_TARGET_ARM_`.

```
#ifdef _ARM_
#define LOG2_PTRSIZE	2 
#define ALIGN_ACCESS    ((1<<LOG2_PTRSIZE)-1) 
#endif

#if defined(ALIGN_ACCESS) && !defined(_MSC_VER) 
... existing code ...
```